modules/openfoam/src/OpenFOAM.f90

```diff
@@ -1004,42 +1005,133 @@ END SUBROUTINE CalcForceActuatorPositionsTower
+   do i=1,p_OpFM%nNodesForceBlade-1
+       pUniform(i) = (i-1)*dRforceNodes
+   end do
+   pUniform(p_OpFM%nNodesForceBlade) = p_OpFM%BladeLength
+   p_OpFM%forceBldRnodes(:) = pUniform(:)
+
+   if (p_OpFM%NodeClusterType .eq. 0) then
+       print*, "Using uniform blade force node clustering."
+      !do i = 1, p_OpFM%nNodesForceBlade
+      !    print*, "r(",i,") = ", pUniform(i)
+      !end do
+   end if
+
+
+   ! If non-uniform spacing is called for, compute the spacing.  
+   ! We know that if the spacing is proportional to chord, then at each
+   ! element, ds = a*c, where a is some constant factor across the entire
+   ! blade.  Therefore sum(ds) = sum(a*c) = L, so a is L/sum(c).  Because
+   ! we don't know c at every point exactly beforehand, we have to
+   ! iterate using c from our previous best guess.  We know things have
+   ! converged when a = c/ds is constant along the blade, so that is our
+   ! convergence check.  We take the difference between a = c/ds between
+   ! all neighboring points to see how different they are, and take the
+   ! rms of that error as the convergence measure (eSum).
+   if (p_OpFM%NodeClusterType .eq. 1) then
+
+       ! For chord-based clustering (increase resolution in regions of decreased chord), an iterative solution to the grid spacing is used.
+       ! The initial guess to the spacing is uniform spacing, so start with that.
+       pNonUniform(:) = pUniform(:)
+
+       ! Get the chord at the initial force points.
+       call OpFM_InterpolateForceNodesChord(initOut_AD, p_OpFM, u_OpFM, ErrStat2, ErrMsg2)
+       cNonUniform(1:p_OpFM%nNodesForceBlade) = u_OpFM%forceNodesChord(2:p_OpFM%nNodesForceBlade+1)
+
+       ! Iterate on a chord-based non-uniform spacing.
+       counter = 0
+       e = 1.0e+6
+       eTol = 1.0e-6
+       eSum = sqrt(sum(e*e))
+       do while ((eSum .gt. eTol) .and. (counter < 50))
+
+          !set the non-uniform spacing to ds = (sum(ds^) / sum(c^)) * c^, where
+          !the ^ denotes from the last iteration.  To begin the iteration, we
+          !use ds = uniform.
+          sNonUniform(:) = (p_OpFM%BladeLength)*cNonUniform(:)/(sum(cNonUniform(2:p_OpFM%nNodesForceBlade-1)) + 0.5*(cNonUniform(1)+cNonUniform(p_OpFM%nNodesForceBlade)))
+
+          ! set the new blade points based on the new ds.
+          do i = 2, p_OpFM%nNodesForceBlade
+             pNonUniform(i) = pNonUniform(i-1) + 0.5*(sNonUniform(i-1) + sNonUniform(i))
+          end do
+          pNonUniform(p_OpFM%nNodesForceBlade) = p_OpFM%BladeLength
+          p_OpFM%forceBldRnodes(:) = pNonUniform(:)
+
+          ! interpolate chord to the new points to get the updated chord values
+          call OpFM_InterpolateForceNodesChord(initOut_AD, p_OpFM, u_OpFM,ErrStat2, ErrMsg2)
+          cNonUniform(1:p_OpFM%nNodesForceBlade) = u_OpFM%forceNodesChord(2:p_OpFM%nNodesForceBlade+1)
+
+          ! compute a = c/ds
+          cByS(:) = cNonUniform(:)/sNonUniform(:)
```

Comment:
@mchurchf here, too, on the `(:)`. There are quite a few places in the new code that has this syntax and should be removed

Reply:
@rafmudaf Okay, I fixed those and tested.  All works identically when the (:) are removed.